yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+        kwargs_test : dict
+            Keyword arguments passed to :py:func:`yasa.plot_hypnogram` when plotting ``hypno_test``.
+
+        Returns
+        -------
+        ax : :py:class:`matplotlib.axes.Axes`
+            Matplotlib Axes
+
+        Examples
+        --------
+        .. plot::
+
+            >>> from yasa import simulate_hypnogram
+            >>> hyp = simulate_hypnogram(seed=7)
+            >>> ax = hyp.evaluate(hyp.simulate_similar()).plot_hypnograms()
+        """
+        assert isinstance(legend, (bool, dict)), "`legend` must be True, False, or a dictionary"
+        assert isinstance(kwargs_ref, dict), "`kwargs_ref` must be a dictionary"
+        assert isinstance(kwargs_test, dict), "`kwargs_test` must be a dictionary"
+        assert not "ax" in kwargs_ref | kwargs_test, (
+            "ax can't be supplied to `kwargs_ref` or `kwargs_test`, use the `ax` keyword instead"
+        )
+        plot_kwargs_ref = {"highlight": None, "alpha": 0.8}
+        plot_kwargs_test = {"highlight": None, "alpha": 0.8, "color": "darkcyan", "ls": "dashed"}
+        plot_kwargs_ref.update(kwargs_ref)
+        plot_kwargs_test.update(kwargs_test)
+        if ax is None:
+            ax = plt.gca()
+        self.hypno_ref.plot_hypnogram(ax=ax, **plot_kwargs_ref)
+        self.hypno_test.plot_hypnogram(ax=ax, **plot_kwargs_test)
+        if legend and "label" in plot_kwargs_ref | plot_kwargs_test:
+            if isinstance(legend, dict):
+                ax.legend(**legend)
+            else:
+                ax.legend()
+        return ax
+
+    def plot_roc(self, palette=None, ax=None, **kwargs):
+        """Plot ROC curves for each stage.
+
+        Parameters
+        ----------
+        palette : dict or None
+            If a dictionary, keys are stages and values are corresponding colors.
+        ax : :py:class:`matplotlib.axes.Axes`
+            Axis on which to draw the plot, optional.
+        **kwargs : key, value pairs
+            Additional keyword arguments are passed to the :py:func:`matplotlib.pyplot.plot` call.
+
+        Returns
+        -------
+        ax : :py:class:`matplotlib.axes.Axes`
+            Matplotlib Axes
+        """
+        raise NotImplementedError("Requires probability/confidence values.")
+
+class SleepStatsEvaluation:
+    """
+    Evaluate agreement between two measurement devices by comparing summary sleep statistics across
+    multiple participants or sessions.
+
+    For example, the reference device might be PSG and the test device might be a wearable device.
+
+    Parameters
+    ----------
+    data : :py:class:`pandas.DataFrame`
+        A pandas dataframe with sleep statistics from two different
+        devices for multiple subjects
+    reference : str
+        Name of column containing the reference device sleep statistics.
+    test : str
+        Name of column containing the test device sleep statistics.
+    subject : str
+        Name of column containing the subject ID.
```

Comment:
Is "subject" too specific here? What if you're looking at multiple sessions from the same participant? Or multiple sessions from multiple participants? I would suggest a more radical approach where the clustering/grouping is defined by the index of ``data``. That way, if you have multiple sessions and participants you can just have a pd.MultiIndex with two levels, and so on.

Reply:
Yes, this was driving me crazy. Wasn't sure what to call it, because it's not always "subject". They're not individual "hypnograms" either, since each row is two different hypnograms (same sleep, different scorer). It's something like individual "sleeps" but obviously that sounds weird. Ideally there would be some word for sleep that functions _only_ as a noun, but all I can think of also work as verbs, which causes the weirdness.

> I would suggest a more radical approach where the clustering/grouping is defined by the index of `data`.

The input dataframes are significantly different than when this comment as made, and now the indices are used. I thought about requiring the user to name the index, which would be used here. I decided against it but could add it in. In any case, the language of "subject" is still annoying because it has to be mentioned a lot in docstrings.

I revised the relevant docstrings language a bit, see what you think and please give advice.